txpipe/source_selector.py

```diff
@@ -489,7 +489,7 @@ def data_iterator(self):
             shear_cols += ["redshift_true"]
 
         chunk_rows = self.config["chunk_rows"]
-        return self.iterate_hdf("shear_catalog", "shear", shear_cols, chunk_rows)
+        return self.iterate_hdf("shear_catalog", "shear", shear_cols, chunk_rows, longest=True)
```

Comment:
This is needed because _1p _1m _2p _2m for a given column have different lengths.

Reply:
Rearranged things and fixed (removed `longest=True`), but the test is still failing because there are no weights_1p weights_1m etc  columns in the test file.